nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -732,64 +845,77 @@ def forward(
         if get_key_value:
             attention_output, presents = attention_output
 
+        # If normformer, apply norm on the output of the self attention.
+        if self.arch == 'normformer':
+            # Normformer normalization
+            attention_output = attention_output + attention_bias if attention_bias is not None else attention_output
+            normalization_output = self.post_attention_normformer_norm(attention_output)
+            attention_bias = None
+
         # Residual connection.
         if self.apply_residual_connection_post_layernorm:
-            residual = layernorm_output
+            residual = normalization_output
         else:
             residual = hidden_states
 
         # jit scripting for a nn.module (with dropout) is not
         # trigerring the fusion kernel. For now, we use two
         # different nn.functional routines to account for varying
         # dropout semantics during training and inference phases.
-        if self.bias and self.bias_dropout_fusion:
-            if self.training:
-                bias_dropout_add_func = bias_dropout_add_fused_train
-            else:
-                bias_dropout_add_func = bias_dropout_add_fused_inference
-        elif self.bias and not self.bias_dropout_fusion:
-            bias_dropout_add_func = get_bias_dropout_add(self.training)
-        else:
-            bias_dropout_add_func = dropout_add
+
+        bias_dropout_add_func = self._get_bias_droput_add_func(arch=self.arch, position_after='attention')
```

Comment:
In the Normformer paper I could not find the dropout after attention. Do we add this here in purpose? (or perhaps I missed it in the paper?)

Reply:
The normformer paper doesn't really include any modifications to where dropout is performed so this part is exactly how it is in the original transformer arch.